File: webApp/src/app/shared/models/order.ts
import { ProductInOrder } from "./productInOrder";
import { User } from "./user";

export class Order {
    id: number = 0;
    producerId: number = 0;
    customerId: number = 0;
    customer: User = new User();
    receivedAt: string = '';
    sentAt: string = '';
    isSent: boolean = false;
    totalPrice: number = 0;
    products: ProductInOrder[] = [];
}

File: api/Models/Order.cs
namespace api.Models
{
    public class Order
    {
        public int Id { get; set; }
        public int ProducerId { get; set; }
        public int CustomerId { get; set; }
        public User? Customer { get; set; }
        public DateTime ReceivedAt { get; set; }
        public DateTime? SentAt { get; set; }
        public bool IsSent { get; set; }
        public float TotalPrice { get; set; }
        public ICollection<ProductInOrder>? Products { get; set; } = null!;
    }
}